New warning on GCC 10

Version of Chocolate Doom: git master a8e8763cba49db50e4a9dc7198c6e4b6dcfbcbfe

Observed behavior: New warning on GCC 10  (not yet released but frozen so warnings should not change)
```
chocolate-doom/src/heretic/mn_menu.c: In function ‘MN_Responder’:
chocolate-doom/src/heretic/mn_menu.c:1529:29: error: writing 1 byte into a region of size 0 [-Werror=stringop-overflow=]
 1529 |                 *textBuffer = ASCII_CURSOR;
      |                             ^
chocolate-doom/src/heretic/mn_menu.c:145:13: note: at offset -1 to object ‘SlotText’ with size 108 declared here
  145 | static char SlotText[6][SLOTTEXTLEN + 2];

```
https://github.com/chocolate-doom/chocolate-doom/blob/a8e8763cba49db50e4a9dc7198c6e4b6dcfbcbfe/src/heretic/mn_menu.c#L1529
False positive or a real error? How do we shut it up?

Also the same warning on hexen https://github.com/chocolate-doom/chocolate-doom/blob/a8e8763cba49db50e4a9dc7198c6e4b6dcfbcbfe/src/hexen/mn_menu.c#L1663

--- a/src/heretic/mn_menu.c
+++ b/src/heretic/mn_menu.c
@@ -1526,9 +1526,9 @@ boolean MN_Responder(event_t * event)
             if (slotptr)
             {
                 *textBuffer = 0;
+                slotptr--;
                 textBuffer--;
                 *textBuffer = ASCII_CURSOR;
-                slotptr--;
             }
             return (true);
         }

--- a/src/heretic/mn_menu.c
+++ b/src/heretic/mn_menu.c
@@ -1527,7 +1527,7 @@ boolean MN_Responder(event_t * event)
             {
                 *textBuffer = 0;
                 slotptr--;
-                textBuffer--;
+                textBuffer = &SlotText[currentSlot][slotptr];
                 *textBuffer = ASCII_CURSOR;
             }
             return (true);

--- a/src/heretic/mn_menu.c
+++ b/src/heretic/mn_menu.c
@@ -1525,7 +1525,8 @@ boolean MN_Responder(event_t * event)
         {
             if (slotptr)
             {
-                *textBuffer-- = 0;
+                *textBuffer = 0;
+                textBuffer--;
                 *textBuffer = ASCII_CURSOR;
                 slotptr--;
             }

--- a/src/hexen/mn_menu.c
+++ b/src/hexen/mn_menu.c
@@ -1660,7 +1660,7 @@ boolean MN_Responder(event_t * event)
             {
                 *textBuffer = 0;
                 slotptr--;
-                textBuffer--;
+                textBuffer = &SlotText[currentSlot][slotptr];
                 *textBuffer = ASCII_CURSOR;
             }
             return (true);

--- a/src/hexen/mn_menu.c
+++ b/src/hexen/mn_menu.c
@@ -1658,7 +1658,8 @@ boolean MN_Responder(event_t * event)
         {
             if (slotptr)
             {
-                *textBuffer-- = 0;
+                *textBuffer = 0;
+                textBuffer--;
                 *textBuffer = ASCII_CURSOR;
                 slotptr--;
             }

--- a/src/hexen/mn_menu.c
+++ b/src/hexen/mn_menu.c
@@ -1659,9 +1659,9 @@ boolean MN_Responder(event_t * event)
             if (slotptr)
             {
                 *textBuffer = 0;
+                slotptr--;
                 textBuffer--;
                 *textBuffer = ASCII_CURSOR;
-                slotptr--;
             }
             return (true);
         }

